chain/memdb/store.go

```diff
@@ -0,0 +1,195 @@
+package memdb
+
+import (
+	"context"
+	"io"
+	"sort"
+	"sync"
+
+	"github.com/drand/drand/chain"
+	"github.com/drand/drand/chain/errors"
+)
+
+// Store represents access to the in-memory storage for beacon management.
+type Store struct {
+	storeMtx *sync.RWMutex
+	store    []*chain.Beacon
+	maxSize  int
+}
+
+// NewStore returns a new store that provides the CRUD based API needed for
+// supporting drand serialization.
+func NewStore() *Store {
+	const maxSize = 10
+
+	return &Store{
+		storeMtx: &sync.RWMutex{},
+		store:    []*chain.Beacon{},
+		maxSize:  maxSize,
+	}
+}
+
+func (m *Store) Len(_ context.Context) (int, error) {
+	m.storeMtx.RLock()
+	defer m.storeMtx.RUnlock()
+
+	return len(m.store), nil
+}
+
+func (m *Store) Put(_ context.Context, beacon *chain.Beacon) error {
+	m.storeMtx.Lock()
+	defer m.storeMtx.Unlock()
+	defer func() {
+		if len(m.store) > m.maxSize {
+			m.store = m.store[len(m.store)-m.maxSize:]
+		}
+	}()
+
+	found := false
+	defer func() {
+		sort.Slice(m.store, func(i, j int) bool {
```

Comment:
this will also resort if the beacon already exists

Reply:
Good catch. Fixed.